orchestration/hca_orchestration/resources/config/datasets.py

```diff
@@ -0,0 +1,84 @@
+import json
+import uuid
+from datetime import datetime
+from typing import Optional
+
+from dagster import InitResourceContext, resource, Array, Field, Noneable
+
+from hca_orchestration.contrib.data_repo.data_repo_service import DataRepoService
+from hca_orchestration.models.hca_dataset import TdrDataset
+from hca_orchestration.support.dates import dataset_snapshot_formatted_date
+
+
+@resource(
+    required_resource_keys={"data_repo_service"},
+    config_schema={
+        "dataset_id": str
+    }
+)
+def passthrough_hca_dataset(init_context: InitResourceContext) -> TdrDataset:
+    """
+    Given a TDR dataset UUID, fetches the dataset info from TDR and returns
+    a fully hydrated TdrDataset object.
+    """
+    data_repo_service: DataRepoService = init_context.resources.data_repo_service
+    target_dataset = data_repo_service.get_dataset(init_context.resource_config["dataset_id"])
+
+    return target_dataset
+
+
+@resource(
+    config_schema={
+        "env": str,
+        "region": str,
+        "policy_members": Array(str),
+        "billing_profile_id": str,
+        "qualifier": Field(Noneable(str), default_value=None, is_required=False),
+    },
+    required_resource_keys={"hca_project_id", "data_repo_service", "run_start_time"}
+)
+def find_or_create_project_dataset(init_context: InitResourceContext) -> Optional[TdrDataset]:
```

Comment:
This is the refactored + renamed `build_new_target_hca_dataset` method. Of note:
* Removed the resource dependency on `hca_project_copying_config` which we were using to get the related hca_project_id in favor of an explicit `hca_project_id` resource dependency.
* Removed the hard check on run start time against the resource context because it wasn't available in unit test. I have factored that out into an additional upstream resource dependency.